src/routes/comments.js

```diff
@@ -84,34 +106,54 @@ Create mongoose ObjectId from parameter
 Find and delete comment in db using findByIdAndDelete()
 Return confirmation message on success
 */
-router.delete("/:commentID", async (req, res) => {
-  Comment.findByIdAndDelete(new mongoose.Types.ObjectId(req.params.commentID))
-    .then(deleted => {
-      if (deleted) {
-        res.json({ deleted: deleted });
-      } else {
-        return Promise.reject("Error: Comment not found");
+router.delete('/:commentID', (req, res) => {
+  Comment.findById(req.params.commentID)
+    .then(comment => {
+      if (!comment) {
+        return res.status(404).json({ Error: 'Comment not found' });
       }
+
+      if (req.user._id !== comment.user.toString()) {
+        return res.status(403).json({ message: "This isn't your comment" });
+      }
+
+      comment.remove().then(() => res.status(204).end());
     })
-    .catch(err => res.status(404).json(err));
+    .catch(err => res.status(500).json('Something went wrong'));
 });
 
 /*
-Route PUT requests 
+Route PATCH requests 
 Create mongoose ObjectId from parameter
 Find and update comment in db using findByIdAndUpdate()
 Return response containing updated comment in JSON on succes
 */
-router.put("/:commentID", async (req, res) => {
-  const { text } = req.body;
-
-  Comment.findByIdAndUpdate(
-    new mongoose.Types.ObjectId(req.params.commentID),
-    { text },
-    { upsert: false, new: true }
-  )
-    .then(comment => res.json(comment))
-    .catch(err => res.status(404).json({ error: "Something went wrong" }));
+router.patch('/:commentID', (req, res) => {
+  const { text: textToUpdate } = req.body;
```

Comment:
I don't understand why you did 'const {text:textToUpdate}=req.body' instead of 'const {text}=req.body'